autopilot/agent.go

```diff
@@ -589,10 +619,23 @@ func (a *Agent) controller() {
 							pub.SerializeCompressed(),
 							err)
 					}
+
+					// Now that we have disconnected, we can
+					// remove this node from our pending
+					// conns map, permitting subsequent
+					// connection attempts.
+					delete(pendingConns, nodeID)
+					pendingMtx.Unlock()
```

Comment:
This will lead to a double unlock as is if the path is executed.

Reply:
It's missing a lock right above.

Reply:
fixed